django/db/models/query_utils.py

```diff
@@ -147,7 +147,9 @@ def __get__(self, instance, cls=None):
             val = self._check_parent_chain(instance)
             if val is None:
                 instance.refresh_from_db(fields=[field_name])
-                val = getattr(instance, field_name)
+                # Now the instance dict contains the reloaded data.
+                # Using getattr() will do extra call to __get__ method.
+                val = data[field_name]
```

Comment:
If `getattr()` is not necessary, do we need this round trip at all?
```python
if val is None:
   ...
    val = data[field_name]
data[field_name] = val
```
It looks that we can remove [this line](https://github.com/django/django/pull/13036/files#diff-1e7fc0d7d1b36358e371fab97bd1ddb1R152) :thinking: .

Reply:
Perfect solution. I will run the tests just to check.

Reply:
The code passed all the tests. Patch updated.
Thank you.